Src/FluentAssertions/CallerStatementBuilder.cs

```diff
@@ -0,0 +1,235 @@
+
+namespace FluentAssertions
+{
+    internal class CallerStatementBuilder
+    {
+        private readonly StringBuilder statement;
+        private readonly IEnumerable<IHandler> handlers;
+
+        public CallerStatementBuilder()
+        {
+            statement = new StringBuilder();
+            handlers = new IHandler[]
+            {
+                new QuotesHandler(statement),
+                new SingleLineCommentHandler(),
+                new MultiLineCommentHandler(statement),
+                new SemicolonHandler(statement),
+                new ShouldCallHandler(statement),
+                new AddNonEmptySymbolHandler(statement)
+            };
+        }
+
+        public Result Append(string symbols)
+        {
+            Result result = Result.InProgress;
+            using var symbolEnumerator = symbols.GetEnumerator();
+            while (symbolEnumerator.MoveNext() && result != Result.Done)
+            {
+                result = Result.InProgress;
+                using var handlerEnumerator = handlers.GetEnumerator();
+                while (handlerEnumerator.MoveNext() && result == Result.InProgress)
+                {
+                    result = handlerEnumerator.Current.Handle(symbolEnumerator.Current);
+                }
+            }
+
+            return result;
+        }
+
+        public override string ToString() => statement.ToString();
+
+        public enum Result
+        {
+            InProgress,
+            Handled,
+            Done
+        }
+
+        private interface IHandler
+        {
+            Result Handle(char symbol);
+        }
+
+        private class QuotesHandler : IHandler
+        {
+            private readonly StringBuilder statement;
+            private char isQuoteEscapeSymbol = '\\';
+            private bool isQuoteContext;
+            private char? previousChar;
+
+            public QuotesHandler(StringBuilder statement) => this.statement = statement;
+
+            public Result Handle(char symbol)
+            {
+                if (symbol == '"')
+                {
+                    if (isQuoteContext)
+                    {
+                        if (previousChar != isQuoteEscapeSymbol)
+                        {
+                            isQuoteContext = false;
+                            isQuoteEscapeSymbol = '\\';
+                        }
+                    }
+                    else
+                    {
+                        isQuoteContext = true;
+                        if (IsAtEscaped())
+                        {
+                            isQuoteEscapeSymbol = '"';
+                        }
+                    }
+                }
+
+                if (isQuoteContext)
+                {
+                    statement.Append(symbol);
+                }
+
+                previousChar = symbol;
+                return isQuoteContext ? Result.Handled : Result.InProgress;
+            }
+
+            private bool IsAtEscaped()
```

Comment:
🔧 As mentioned before, I would rename this to `IsVerbatim`

Reply:
Done